internal/modifier/gated.go

```diff
@@ -78,5 +79,10 @@ func NewFeatureGatedModifier(logger logger.Interface, cfg *config.Config, image
 		discoverers = append(discoverers, d)
 	}
 
+	if !cfg.Features.AllowCUDACompatLibsFromContainer.IsEnabled() {
```

Comment:
Maybe I am confused, but shouldn't this be negated? I would expect this hook to be added only when `allow-cuda-compat-libs-from-container` is enabled.

Reply:
The purpose of this hook is to provide the same functionality as is currently in `libnvidia-container`. This means that if we disable that (as we do by default in this branch), we would have to inject the hook. If a user opts-in to the older behaviour, then the hook is not required.

Reply:
Thanks for the clarification.